common/djangoapps/tedix_ro/views.py

```diff
@@ -173,73 +179,91 @@ def user_due_date_data(user, due_date_datetime):
-                        'classrom': students_class,
-                        'average_score': '{}%'.format(average_score * 100) if average_score is not None else 'n/a',
-                    })
+                ).values_list('student__user_id', flat=True)
+
+                students_in_class_count = students_in_class.count()
+                count_answers_first_attempt = 0
+                for student in students_in_class:
+                    count_answers_first_attempt += get_count_answers_first_attempt(student, course_key)[0]
+
+                average_score = 'n/a'
+                if count_answers_first_attempt:
+                    average_score = '{}%'.format(
+                        (float(count_answers_first_attempt)/students_in_class_count) / all_questions_count * 100
+                    )
+
+                data.append({
+                    'course_name': course_name,
+                    'report_link': course_report_link,
+                    'due_date': course_due_date_data,
+                    'classroom': students_class,
+                    'average_score': average_score,
+                })
 
         if hasattr(user, 'studentprofile'): 
-            student_report = StudentReportSending.objects.filter(
-                course_id=course_key,
-                user=user
-            ).first()
-            if not student_report:
-                continue
             student_class = user.studentprofile.classroom
             course_due_date = StudentCourseDueDate.objects.filter(
                 student=user.studentprofile, 
                 course_id=course_key,
             ).first()
 
+            count_answers_first_attempt, answered_questions_count = get_count_answers_first_attempt(user, course_key)
+            average_score = 'n/a'
+            if count_answers_first_attempt:
+                average_score = '{}%'.format(
+                    count_answers_first_attempt * 100 / all_questions_count
+                )
+
+
             if course_due_date:
                 course_due_date_data = user_due_date_data(user, course_due_date.due_date)
 
-            data.append({
-                'course_name': course_name,
-                'report_link': course_report_link,
-                'due_date': course_due_date_data,
-                'classrom': student_class,
-                'average_score': '{}%'.format(student_report.grade * 100) if student_report else 'n/a',
-            })
+                data.append({
+                    'complete': bool(answered_questions_count == all_questions_count),
+                    'course_name': course_name,
+                    'report_link': course_report_link,
+                    'due_date': course_due_date_data,
+                    'classroom': student_class,
+                    'average_score': average_score,
+                })
+
     context = {
         'data': data,
-        'is_student': hasattr(user, 'studentprofile')
+        'user': user,
```

Comment:
you can get a user from the request in a makotemplate, no need to pass user via context.